src/puppet.js

```diff
@@ -89,7 +89,142 @@
 
   }
 
-  function PuppetNetworkChannel(puppet, remoteUrl, useWebSocket, onReceive, onSend, onError, onStateChange) {
+  /**
+   * @constructor
+     */
+  function Reconnector(eventDispatcher, reconnectAction) {
+    this._eventDispatcher = eventDispatcher;
+    this._reconnect = reconnectAction;
```

Comment:
I'd make `eventDispatcher` and `reconectAction` public members, as they are given by public API anyways, and I see no harm in changing them in runtime.